### PR TITLE
feat!: VideoPlayerProvider

### LINKED ISSUE #87: Stats: why do you need to pass all the props
Hello,

1. As a developer I'm wondering why all these props are need to be passed `Stats` component:
Screenshot 2020-12-09 at 19 53 03

IMO all these props should already be present in the player and be accessible via `state`, `context API`, `recoil`, osv and should not be passed to the component to prevent the player and stats to be not in sync.

2. IMO `api` is confusing name for the prop also and IMO you should have comments above the props types in order to display hints in VS Code and help the developers to understand what the prop does, e.g.:,

Screenshot 2020-12-10 at 12 32 16
Screenshot 2020-12-10 at 12 32 25

### PR DESCRIPTION
**Pull request by lekoaf**
Implements a VideoPlayer.Provider so you don't have to shuffle around state throughout the components.

Fixes #87

### CODE REVIEW

**Comment by lekoaf** [comment]
Just reread @steabert's comments in the issue.
> Lower-level components should probably keep the props (and not depend on the context).

So, maybe I went a bit over board with this then. I'll reset the "low level" video components if you wish to.

File: lib/PlaybackArea.tsx [hunk 1]
```diff
@@ -157,16 +153,17 @@ const search = (api: string, parameters: VapixParameters = {}) => {
...
-  play,
-  refresh,
   onPlaying,
   onSdp,
   metadataHandler,
   secure = window.location.protocol === 'https',
 }) => {
+  const { host, api, parameters, refresh } = useContext(VideoPlayerContext)
+
+  if (api === undefined) {
```
Comment (lekoaf) [inline lib/PlaybackArea.tsx:163]: So, this one stumped me. For _no_ apparent reason, `api` was `undefined` in `Player.tsx` but worked just fine in `BasicPlayer.tsx`. I'd love for someone to help me find out why that is. I tried for 30 minutes before I thought "f it" and implemented this guard.

**Comment by lekoaf** [comment]
> Just reread @steabert's comments in the issue.
> 
> > Lower-level components should probably keep the props (and not depend on the context).
> 
> So, maybe I went a bit over board with this then. I'll reset the "low level" video components if you wish to.

I removed the context from the lower level stuff.

File: lib/PlaybackArea.tsx (hunk 1 above)
Comment (steabert) [reply lib/PlaybackArea.tsx:165]: If you rebase this on the latest commit, `api` is gone (and replaced by a stricter typed `format`). Maybe that will solve the issue?

**Pull request closed without merging**